aten/src/ATen/native/cuda/MultiTensorApply.cuh

```diff
@@ -268,7 +271,7 @@ void multi_tensor_apply(
         loc_block_info = 0;
         if (chunk == chunks - 1) {
           loc_tensor_info = 0;
-          tensorListMeta.start_tensor_this_launch = t + 1;
+          tensorListMeta.start_tensor_this_launch = processed;
```

Comment:
Ah, I see this is used for the reduce ops, and once we filtered out the 0dim ones, the offset becomes off (processed is # processed, not the index of which tensor we're on (so it's smaller)). Thanks for the fix! Would you have time to do a test case in test_foreach.py? 

If not, I can add one tomorrow

Reply:
Added testing